controllers/contentlibrary/v1alpha2/clustercontentlibraryitem/clustercontentlibraryitem_controller.go

```diff
@@ -156,25 +156,43 @@ func (r *Reconciler) ReconcileNormal(ctx *context.ClusterContentLibraryItemConte
 			savedStatus = cvmi.Status.DeepCopy()
 		}()
 
+		// Resetting the conditions to recalculate the latest state.
+		cvmi.SetConditions([]metav1.Condition{})
```

Comment:
This will cause the image object to be updated on every reconcile because we'll always set a new (current) LastTransitionTime. The condition helper Set() would otherwise preserve that unless there is a state transition.

Reply:
I had specifically done this to remove the older conditions which might be present in case of brownfield environments. Having those v1a1 conditions is alright I suppose, should not be a big issue. Going forward, we are planning to rally around the `Ready` condition only, so the presence of those conditions in some cases should not be a problem.

Reply:
I'd agreed the older conditions are probably harmless. If not we can use the version conversion to remove them. But we do need to preserve the LastTransitionTime if there wasn't a state transition.